acceptance/tests/windows/prevent_password_changes.rb

```diff
@@ -0,0 +1,104 @@
+require 'date'
+
+def random_username
+  "pl#{rand(999999).to_i}"
+end
+
+def change_password_manifest(username)
+  return <<-MANIFEST
+    user { '#{username}':
+      ensure   => 'present',
+      password => 'Password-#{rand(999999).to_i}'
+    }
+  MANIFEST
+end
+
+def assert_password_matches_on(host, username, password)
```

Comment:
I have similar code here: https://github.com/ekinanp/puppet/blob/PUP-5216-rebased/acceptance/lib/puppet/acceptance/windows_utils.rb#L60 that's used e.g. here: https://github.com/ekinanp/puppet/blob/PUP-5216-rebased/acceptance/tests/resource/user/should_manage_windows_passwords.rb#L62

Would be good to agree on which password-assertion-style should be used so I can update my PR accordingly. I don't care on my end.

If we do go with `assert_password_matches_on`, then I think it would be good to have the assertion message be a parameter, and move it over to the `windows_util.rb` module.

Reply:
I didn't know that another method existed to do this - I will update.

Reply:
It doesn't exist yet -- that will be added if/when https://github.com/puppetlabs/puppet/pull/7115 gets merged.

Reply:
It seems like there is some overlap between these PRs. This is my first time contributing to this repo, so I'm thinking I will defer to whatever is decided in #7115 and revise this PR later.

Reply:
That's probably not going to be merged for a while, not until after Puppetize. I'm fine with updating that PR with what's decided here too if you would like to move this along.

Reply:
Ok, I think I'll look at your PR and match this one to it